src/date/format.js

```diff
@@ -97,7 +96,12 @@ return function( date, numberFormatters, properties ) {
 			// Quarter
 			case "Q":
 			case "q":
-				ret = Math.ceil( ( date.getMonth() + 1 ) / 3 );
+
+        // There's no good way to do this with a generalized date.
+				// We have to approximate this, assuming a 12-month year and
+				// month numbers that correspond to the correct period of the year
+				ret = Math.min( parseInt( gdate.getMonth(), 10 ), 12 );
```

Comment:
Please, what's the return value of `.getMonth()` is it a string? Where does this string comes from? I guess this comes from CLDR, the key of months object. Just confirming. Is this always a digit?